src/link-manager.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Array<genotet.View>>} */
+genotet.linkManager.linkViews = {};
+
+/**
+ * Object that stores the link actions and responses type.
+ */
+/** @type {!Object<string>} */
+genotet.linkManager.linkActions = {};
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.data.VIEW_TYPES.forEach(function(type) {
```

Comment:
This is incorrect. The key of links should be views rather than view types. Recall that it is a directed graph where views are nodes. You need to store the incident edges for nodes, NOT for types of nodes.